### PR TITLE
Improve logging for hostfxr.dll not found

### PR DESCRIPTION
**Pull request by danmosemsft**
Currently if you try to run dotnet.exe out of the corefx bin folder, it will give a message like: `A fatal error occurred, the required library hostfxr.dll could not be found at C:\git\corefx\bin\runtime\netcoreapp-Windows_NT-Release-x64\`. 

This is confusing because `hostfxr.dll` DOES exist in that folder. I assume it is intentional that it doesn't attempt to load it from there..

Now it gives a message like
`A fatal error occurred, the folder C:\git\corefx\bin\testhost\netcoreapp-Windows_NT-Release-x64\host\fxr does not exist`

The change is quite small, the diff is poor.

### CODE REVIEW

**Comment by danmosemsft** [comment]
Also found some cases where things other than c style strings were getting passed to printf. Perhaps logging should use iostream.

**Comment by danmosemsft** [comment]
This logging may also help some mysterious cases like https://github.com/dotnet/corefx/issues/24763

File: src/corehost/corehost.cpp [hunk 1]
```diff
@@ -89,44 +89,57 @@ pal::string_t resolve_fxr_path(const pal::string_t& own_dir)
...
+        fx_ver_t fx_ver(-1, -1, -1);
+        if (fx_ver_t::parse(ver, &fx_ver, false))
         {
-            trace::info(_X("Resolved fxr [%s]..."), ret_path.c_str());
-            return ret_path;
+            max_ver = std::max(max_ver, fx_ver);
         }
     }
 
+    if (max_ver.get_major() == -1)
```
Comment (steveharter) [inline src/corehost/corehost.cpp]: Alternate approach is `== fx_ver_t(-1, -1, -1)`

File: src/corehost/corehost.cpp [hunk 2]
```diff
@@ -89,44 +89,57 @@ pal::string_t resolve_fxr_path(const pal::string_t& own_dir)
...
+    {
+        trace::error(_X("A fatal error occurred, the folder %s does not contain any version-numbered child folders"), fxr_dir.c_str());
```
Comment (steveharter) [inline src/corehost/corehost.cpp]: Can you add [] around the various %s added here. The existing code is inconsistent, but at least new cases should have it.

Comment (danmosemsft) [reply src/corehost/corehost.cpp]: I'll add it around the paths (although I think it makes it harder to read..)

**Comment by danmosemsft** [comment]
@dotnet-bot test Tizen armel Release Build (hung)

**Comment by danmosemsft** [comment]
@dotnet/dnceng the OSX job has been "archiving artefacts" for nearly half an hour. Also the Tizen armel job is not starting...

**Comment by karajas** [comment]
Looking into the old OSX logs looks like the average archive time is about 30 min. This one took slightly longer.
Looking into the Tizen leg

**Comment by karajas** [comment]
CoreCLR had a sudden burst in request that test all of CI as far as I can tell. The machines are taking work (the ubuntu1404-20170120 queue is at 100%)

**Comment by danmosemsft** [comment]
@karajas thanks! @weshaggard should it surprise me that CI jobs in core-setup spend 30 minutes archiving things? What needs to be archived?

**Comment by weshaggard** [comment]
>@weshaggard should it surprise me that CI jobs in core-setup spend 30 minutes archiving things? What needs to be archived?

That sounds wrong and should be investigated.

**Comment by mmitche** [comment]
@weshaggard Most likely it's just data transfer in/out of the dtap lab is slow.  The total artifact size for that was ~329MB.

**Comment by danmosemsft** [comment]
@mmitche that's about 1Mb/s. Why isn't it nearer 1Gb/s. Is there a malfunction?

**Comment by mmitche** [comment]
@danmosemsft Not arguing on the low end.  We have had problems in the past though.  The pipe is shared in/out of the lab.  I'll file an issue.

**Comment by mmitche** [comment]
https://github.com/dotnet/core-eng/issues/2154

**Pull request merged**